waiter/src/waiter/process_request.clj

```diff
@@ -242,14 +242,11 @@
       (try
         (loop [unreported-bytes-to-statsd 0]
           (let [available-bytes (.available input-stream)
-                ;; get a buffer size between buffer-increment-size and max-buffer-size
+                ;; get a buffer size between min-buffer-size and max-buffer-size
                 buffer-size (-> available-bytes
-                              (max 0)
-                              double
-                              (/ buffer-increment-size)
-                              int
-                              inc
-                              (* buffer-increment-size)
+                              (max min-buffer-size)
+                              (bit-shift-right buffer-increment-power-of-2)
+                              (bit-shift-left buffer-increment-power-of-2)
```

Comment:
I think you could use [Integer/highestOneBit](https://docs.oracle.com/javase/10/docs/api/java/lang/Integer.html#highestOneBit(int)) here, replacing these bit shifts as well as the `Math/log` calls above.

Reply:
The goal isn't a power of 2.

`(Integer/highestOneBit 7000) => 4096` vs `(-> 7000 (bit-shift-right 10) (bit-shift-left 10)) => 6144`

Reply:
Ah, I misread and thought the Math/log was computed on available-bytes. Masking off the last 9 bits would be easier to understand (and faster) than shift-right followed by shift-left.

Reply:
Done.